### DISCUSSION TITLE
The best way to add disk in a server device

### DISCUSSION THREAD

**Discussion by aivorra** [netbox-community/netbox#17849] [Q&A]
Hi, I'm trying to understand what is the best way to add hot-swap disk in a server device type. I'm not sure if I need to add "device bays", "module bays" or use "inventory items".
In my case the server has 24 bays and I'm using 10 of them with disks.

BR.

**Comment by betadrome** [comment]
> Hi, I'm trying to understand what is the best way to add hot-swap disk in a server device type. I'm not sure if I need to add "device bays", "module bays" or use "inventory items". In my case the server has 24 bays and I'm using 10 of them with disks.
> 
> BR.

I use inventory items for disks because they can be counted very well with the inventory plugin. Device bays are usefull for switch chassis with 2 or more device slots and module bays are used for network cards, etc.

**Reply by candlerb** [reply]
A switch chassis with line cards should use module bays, because the components on a module are integrated into the device.

Device bays should only be used for cases where the child devices are completely independent from the parent, such as:
* blade servers
* mounting brackets (which hold multiple servers in the same block of rack U space)

**Reply by FlorianHeigl** [reply]
> > Hi, I'm trying to understand what is the best way to add hot-swap disk in a server device type. I'm not sure if I need to add "device bays", "module bays" or use "inventory items". In my case the server has 24 bays and I'm using 10 of them with disks.
> > BR.
> 
> I use inventory items for disks because they can be counted very well with the inventory plugin. Device bays are usefull for switch chassis with 2 or more device slots and module bays are used for network cards, etc.

I understand your reasoning and just looked at this - but what do you do with the "Name" for disks?

Another (good) thing is with inventory items I don't need to consider stuff like hotswap or not.
I also really dislike to have the manufacturers in the main manufacturers list, but there'll be no avoiding that.

**Reply by Serverhamster** [reply]
Is that suggestion still valid? I read the following in the [documentation](https://netboxlabs.com/docs/netbox/models/dcim/inventoryitem/).

> Beginning in NetBox v4.3, **the use of inventory items has been deprecated**. They are planned for removal in a future NetBox release. Users are strongly encouraged to begin using [modules](https://netboxlabs.com/docs/netbox/models/dcim/module/) and [module types](https://netboxlabs.com/docs/netbox/models/dcim/moduletype/) in place of inventory items. Modules provide enhanced functionality and can be configured with user-defined attributes.
